src/Orleans.Runtime/Core/Dispatcher.cs

```diff
@@ -164,8 +168,8 @@ public void ReceiveMessage(Message message)
                             {
                                 try
                                 {
-                                    await this.localGrainDirectory.UnregisterAfterNonexistingActivation(
-                                        nonExistentActivation, origin);
+                                    logger.Trace("UnregisterAfterNonexistingActivation addr={0} origin={1}", nonExistentActivation, origin);
```

Comment:
This trace message needs a guard

Reply:
Also, prefer to use LogTrace instead of our string.format-based shim

Reply:
I just restored a log line that was wrongly deleted in the multi-cluster PR. I will update it